src/routing_node.rs

```diff
@@ -138,6 +138,12 @@ impl RoutingNode {
                 Ok(Action::DropConnections(connections)) => {
                     self.drop_connections(connections);
                 },
+                Ok(Action::ExpectedConnection(expected_connection)) => {
```

Comment:
Can the action include the actual verb "MatchExpectedConnection" and similarly for the other? Same for function names